### PR TITLE
Fixed #18823: Prefix dropdown in checkout doesn't select empty value by defau…

### LINKED ISSUE #18823: Prefix dropdown in checkout doesn't select empty value by default if required
### Preconditions (*)

1. Magento 2.2.6
2. Luma Theme

### Steps to reproduce (*)

1. In Admin Backend go to Stores > Configuration > Customers > Name and Address Options
2. Set "Show Prefix" to "Required"
3. Set "Prefix Dropdown Options" to ";Mr.;Mrs."
4. Save Config
5. Go to your store
6. Add a item to your cart
7. Go to checkout

### Expected result (*)

I expect the empty value in the prefix dropdown to be selected by default.
prefix_expected

### Actual result (*)

Instead of having the empty value selected by default it will be always the first value selected.
prefix_actual

### Additional information
issue_18823

### PR DESCRIPTION
**Pull request by mikebox**
Fixed #18823: Prefix dropdown in checkout doesn't select empty value by default if required

### Description (*)
Added a condition to ensure the options provided by the customers start with the separator(;). If the separator exists at the start of the options, adding an empty option in the select.

Additional Notes:
array_filter() removes the empty characters. Removing that might create additional problems of having empty options in the middle. So added a condition to ensure the starting character and act based on that.

### Fixed Issues (if relevant)
1. magento/magento2#18823: Prefix dropdown in checkout doesn't select empty value by default if required

### Manual testing scenarios (*)

1. In Admin Backend go to Stores > Configuration > Customers > Name and Address Options
2. Set "Show Prefix" to "Required"
3. Set "Prefix Dropdown Options" to ";Mr.;Mrs."
4. Save Config
5. Go to your store
6. Add a item to your cart
7. Go to checkout

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by lenaorobei** [comment]
Related to https://github.com/magento/magento2/pull/20896

**Comment by lenaorobei** [comment]
As per discussion with PO @mbrinton01, the expected behavior is:
`Mr;Ms;`  - select with `Mr` , `Ms` and `""`
`;Mr;Ms` - select with `""`, `Mr` and `Ms`
`;Mr;Ms;`- select with `""`, `Mr`, `Ms` and `""`

https://github.com/magento/magento2/issues/17861 was configuration issue but not the functional.

File: app/code/Magento/Customer/Model/Options.php [hunk 1]
```diff
@@ -98,6 +98,11 @@ private function prepareNamePrefixSuffixOptions($options, $isOptional = false)
             return false;
         }
         $result = [];
+
+        if (substr($options, 0, 1) === ";") {
```
Comment (lenaorobei) [inline app/code/Magento/Customer/Model/Options.php]: Please remove this check.

File: app/code/Magento/Customer/Model/Options.php [hunk 2]
```diff
@@ -98,6 +98,11 @@ private function prepareNamePrefixSuffixOptions($options, $isOptional = false)
...
+            $result = array_merge([' ' => ' '], $result);
+        }
+
         $options = array_filter(explode(';', $options));
```
Comment (lenaorobei) [inline app/code/Magento/Customer/Model/Options.php]:
```suggestion
        $options = explode(';', $options);
```

**Review by lenaorobei — CHANGES_REQUESTED** [review]
@mikebox please see my comments and consider covering these cases with MFTF tests. Thank you.

**Comment by swnsma** [comment]
Hi @mikebox @lenaorobei 
Could we use a little bit complex but more obvious approach called 'Dynamic Rows'?
https://devdocs.magento.com/guides/v2.3/ext-best-practices/tutorials/dynamic-row-system-config.html

**Comment by lenaorobei** [comment]
I like this idea! Let me confirm with @mbrinton01 since it will be a pretty big UI change and I believe we should take care of `suffix` as well.

**Comment by mbrinton01** [comment]
I'm fine with this approach as it should return the same experience for the user.

**Comment by engcom-Echo** [comment]
I will take care of test coverage.

**Review by lenaorobei — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-6991](https://jira.corp.magento.com/browse/ENGCOM-6991) has been created to process this Pull Request

**Review by engcom-Alfa — COMMENTED** [review]
.

**Review by engcom-Alfa — CHANGES_REQUESTED** [review]
Hi @mikebox.

During testing, we faced the issue.

**_Problem:_** Prefix dropdown in checkout doesn't select an empty value by default

**Manual testing scenario:**

1. In Admin Backend go to Stores > Configuration > Customers > Name and Address Options;
2. Set "Show Prefix" to "Required";
3. Set "Prefix Dropdown Options" to ";Mr.;Mrs.";
4. Save Config and go to your store;
5. Add an item to your cart and go to checkout;

**Actual Result:** The value "**Mr.**" is selected by default
after2

@mikebox Could you take a look?

Thanks!

**Comment by lenaorobei** [comment]
Closing this PR due to inactivity. Please feel free to reopen if you are going to continue working on it.
Thank you.

**Pull request closed without merging**